src/storage/src/hummock/sstable/block.rs

```diff
@@ -29,14 +31,138 @@ pub const DEFAULT_BLOCK_SIZE: usize = 4 * 1024;
 pub const DEFAULT_RESTART_INTERVAL: usize = 16;
 pub const DEFAULT_ENTRY_SIZE: usize = 24; // table_id(u64) + primary_key(u64) + epoch(u64)
 
+#[allow(non_camel_case_types)]
+#[derive(Clone, Copy, PartialEq, Debug)]
+pub enum KeyPrefixLenType {
+    u8 = 1,
+    u16 = 2,
+    u32 = 3,
+}
+
+#[macro_export]
+macro_rules! for_all_type {
+    ([$($arg:ident, )*], $macro:ident) => {
+        $macro! {
+            $($arg, ) *
+            {u8, u8},
+            {u8, u16},
+            {u8, u32},
+            {u16, u8},
+            {u16, u16},
+            {u16, u32},
+            {u32, u8},
+            {u32, u16},
+            {u32, u32}
```

Comment:
nits: I still think listing out all combinations seems unneccessary compared to just impl the relevant functions with three match arms.

this is not a big deal since in either cases the code logic are similar so I am okay with either implementation.